include/igl/principal_curvature.cpp

```diff
@@ -448,7 +448,8 @@ IGL_INLINE void CurvatureCalculator::getKRing(const int start, const double r, s
   int bufsize=vertices.rows();
   vv.reserve(bufsize);
   std::list<std::pair<int,int> > queue;
-  bool* visited = (bool*)calloc(bufsize,sizeof(bool));
+  std::vector<bool> visited;
```

Comment:
Can you initialize it to false as well? Using the constructor `std::vector<bool> visited(bufsize, false);`